### DISCUSSION TITLE
Certification path not found

### DISCUSSION THREAD

**Discussion by likhoded** [vpnhood/VpnHood#175] [Q&A]
Hi, I made a build of VPNHodd.Client.Android is one of the samples and it works successfully. However, I tried to use exactly the same code, without changing anything, in a new project and encountered an error when connecting
`java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.`

What is the reason for this and how can I solve this problem?

**Comment by trudyhood** [comment]
Hi,
Have you tried to connect to our public server or your own server?
I am not sure what you have done. try to compare with our sample and share it with us if you found the issue.

**Reply by likhoded** [reply]
> Привет, Вы пытались подключиться к нашему общедоступному серверу или к вашему собственному серверу? Я не уверен, что вы сделали. попробуйте сравнить с нашим образцом и поделиться им с нами, если вы обнаружили проблему.

I am connecting to the server that the client from the examples is connecting to. I copied the code from the MainActivity in the examples and moved it to the MainActivity of the new project without changing anything. At the same time, the client from the examples successfully connects, but my client does not.

**Comment by trudyhood** [comment]
Can you share your project with support@vpnhood.com?
You can share via Google Drive or anything else.

**Reply by likhoded** [reply]
Sent the project to the specified email address

**Reply by trudyhood** [reply]
Sorry, after a long time, We were too busy. Have you solved the issue?

**Reply by likhoded** [reply]
Hello, you have not responded to the project I sent. But I found out that the cause of the error is a MAUI bug related to certificates and at the moment there is no way to use VPNHOOD in MAUI.

**Reply by likhoded** [reply]
Although in general, if we exclude specifically the problem with certificates, VPN Hood is possible to run in MAUI

**Comment by trudyhood** [comment]
Have you tried this?
https://stackoverflow.com/questions/54715317/xamarin-android-issue-connecting-via-https-to-site-with-self-signed-certificate

VpnHood uses the following code to verify the certificate manually
![image](https://user-images.githubusercontent.com/72891124/208566839-7d951c5c-1f6d-4f61-a6b3-76c79d38394e.png)

Let me know if you find a workaround for it. Some people said android manages it internally but I am wondering how it works for non MAUI projects!
